cmd/skopeo/proxy.go

```diff
@@ -770,23 +896,31 @@ func (buf replyBuf) send(conn *net.UnixConn, err error) error {
 		PipeID:  buf.pipeid,
 	}
 	if err != nil {
+		replyToSerialize.ErrorCode = mapProxyErrorCode(err)
 		replyToSerialize.Error = err.Error()
+
 	}
 	serializedReply, err := json.Marshal(&replyToSerialize)
 	if err != nil {
 		return err
 	}
-	// We took ownership of the FD - close it when we're done.
+	// We took ownership of these FDs, so close when we're done sending them or on error
 	defer func() {
```

Comment:
(Non-blocking, this is pre-existing: this `defer` should happen before we exit on the failure above.)

Reply:
Thanks, I will fix that in a separate PR for clarity

Reply:
Eh that'd be a conflict, so added another commit here